home/tests/test_api.py

```diff
@@ -462,93 +464,95 @@ def test_detail_view_no_content_page(self, uclient):
+    Test the WhatsApp specific functionality of ContentPage like buttons templates and
+    variations
     """
 
+    def create_content_page(
+        self,
+        buttons=None,
+        whatsapp_template_category=None,
+        whatsapp_template_name=None,
+        variation_messages=None,
+    ):
+        """
+        Helper function to create pages needed for each test.
+
+        Parameters
+        ----------
+        buttons : [NextBtn | PageBtn]
+            List of buttons to add to the content page.
+        whatsapp_template_category : str
+            Category of the WhatsApp template.
+        whatsapp_template_name : str
+            Name of the WhatsApp template
+        variation_messages : [VarMsg]
+            Variation messages added to the WhatsApp content block
+        """
+        title = "default page"
+        home_page = HomePage.objects.first()
+        main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
+
+        content_page = PageBuilder.build_cp(
+            parent=main_menu,
+            slug=title.replace(" ", "-"),
+            title=title,
+            bodies=[
+                WABody(
+                    title,
+                    [
+                        WABlk(
+                            "Test WhatsApp Message 1",
+                            buttons=buttons or [],
+                            variation_messages=variation_messages or [],
+                        )
+                    ],
+                )
+            ],
+            whatsapp_template_category=whatsapp_template_category,
+            whatsapp_template_name=whatsapp_template_name,
+        )
+        return content_page
+
     def test_whatsapp_detail_view_with_button(self, uclient):
         """
         Next page buttons in WhatsApp messages are present in the message body.
         """
-        page = ContentPage(
-            title="test",
-            slug="text",
-            enable_whatsapp=True,
-            whatsapp_body=[
-                {
-                    "type": "Whatsapp_Message",
-                    "value": {
-                        "message": "test message",
-                        "buttons": [
-                            {"type": "next_message", "value": {"title": "Tell me more"}}
-                        ],
-                    },
-                }
-            ],
-        )
-        homepage = HomePage.objects.first()
-        homepage.add_child(instance=page)
-        page.save_revision().publish()
+        page = self.create_content_page(buttons=[NextBtn("Tell me more")])
 
         response = uclient.get(f"/api/v2/pages/{page.id}/?whatsapp=true&message=1")
         content = response.json()
         [button] = content["body"]["text"]["value"]["buttons"]
         button.pop("id")
         assert button == {"type": "next_message", "value": {"title": "Tell me more"}}
 
-    def test_whatsapp_template(self, uclient):
```

Comment:
I merged this one into `test_whatsapp_template_fields`, better to just check all the template fields in one go.